documentation.md

```
@@ -0,0 +1,87 @@
+
+*   A C++17 compliant compiler (e.g., GCC, Clang)
+*   Make
+*   [Premake5](https://raw.githubusercontent.com/premake/premake-core/refs/heads/master/BUILD.txt)
+
+_Note for Windows users: While this project might compile on Windows with appropriate tooling (like MinGW/MSYS2 Make or generating Visual Studio solutions via `premake5 vs20XX`), these instructions focus on Linux/macOS. You'll need to adapt the build and run commands accordingly - deal with the pain yourself._
+
+## Building the Project
+
+1.  **Generate Build Files:**
+    Open a terminal in the project's root directory and run Premake to generate the necessary Makefiles:
+    ```bash
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should be surrounded by blank lines
null

(MD031, blanks-around-fences)